File: FlaskServer/controllers/ResponsibleController.py
from flask import Blueprint, jsonify, make_response
from http import HTTPStatus
import logging
from services.ResponsibleService import ResponsibleService

responsible_bp = Blueprint('responsible', __name__, url_prefix='/responsible')


@responsible_bp.route('/get-all')
def get_all():
    service = ResponsibleService()
    try:
        result = service.get_all()
        return make_response(jsonify(result), HTTPStatus.OK)
    except Exception as e:
        logging.error("Error fetching responsible", e)
        return make_response(jsonify({'error': str(e)}), HTTPStatus.INTERNAL_SERVER_ERROR)

File: evangelism-assistant-server/src/main/kotlin/com/quenazapps/evangelism_assistant/controller/ResponsibleController.kt
package com.quenazapps.evangelism_assistant.controller

import com.quenazapps.evangelism_assistant.service.ResponsibleService
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus.*
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/responsible")
class ResponsibleController(
    private val service: ResponsibleService
) {

    companion object {
        private val log = LoggerFactory.getLogger(ResponsibleController::class.java)
    }

    @GetMapping("/get-all")
    fun getAll(): ResponseEntity<Any> {
        return try {
            val result = service.findAll()
            ResponseEntity.status(OK).body(result)
        } catch (e: Exception) {
            log.error("Error fetching responsible", e)
            ResponseEntity.status(INTERNAL_SERVER_ERROR)
                .body(mapOf("error" to e.message))
        }
    }
}